projects/js-toolkit/packages/js-toolkit-core/src/project/liferayCli/Project.ts

```diff
@@ -61,7 +63,31 @@ export default class Project {
 			self.mainModuleFile = this.srcDir.join('index.js');
 		}
 
-		const liferayJson = this._loadLiferayJson();
+		let liferayJson = this._loadLiferayJson();
+
+		const clientExtensionYamlPath = this.dir.join('client-extension.yaml')
+			.asNative;
+
+		if (fs.existsSync(clientExtensionYamlPath)) {
+			try {
+				const yamlConfig = yaml.load(
+					fs.readFileSync(
+						this.dir.join('client-extension.yaml').asNative,
+						'utf8'
+					)
+				);
+
+				liferayJson = merge.all([
+					liferayJson,
+					this._normalizeClientExtensionYaml(yamlConfig),
```

Comment:
Maybe we should warn about duplicates... I've seen tools doing it (for example `tsc` checkes all JSON files against their schemas) and sometimes saved me a lot of headaches.

No need to do it now if you don't want to, but we may once all this stuff is settled. Otherwise people can get very confused if they see the same thing in two files, for example. Or even worse, if they are only looking at one and forgetting about the other :grimacing: .

Reply:
Yes and no... Because I doubt we can do everything with just the yaml file :thinking: We may need more options I mean, and until we have a proper blade team in place I don't think it will be easy for us to put them in the yaml file. 

But yeah, it would probably be a good ending point: putting everything in the yaml file.